### ISSUE TITLE
A kind of Non-English Characters Display Error

### ISSUE THREAD

**Issue by MintYale** [Eugeny/tabby#6479]
**Describe the problem**:
If Tabby is used for Git, then there is most likely  some other non-English characters inputted in(the git commit info, for instance),
so, I have found this display error in Tabby: 
input Chinese or Japanese  without moving cursor by arrow-keys(← and → in the keyboard), everything is OK, like this:
![image](https://user-images.githubusercontent.com/49377524/169706321-18ea728d-66ca-4091-b599-4306867330ef.png)
Yet once moving cursor by arrow-keys, especially move to left(backward), there will be some vertical lines on every Chinese or Japanese Character like this(in my example pic the number of Chinese/Japanese Characters are both 4): 
![image](https://user-images.githubusercontent.com/49377524/169706364-5aebaeb9-dcaa-4965-b54f-e26debe2e54c.png)

This doesn't shows in other terminal apps  such as Powershell/Git Bash/Extraterm on the same PC with the same OS.
This shows on both Windows 10/11, Fedora 36, Ubuntu 22.04 LTS.
This shows no matter what kind of font you are using. 
This has shown in many editions of Tabby.(Since it's not a very serious problem, I didn't report it immediately once I found it, and after that I almost forget it. Recently I got a Windows 11 laptop and I install Tabby on it, then I found this again.)

**To Reproduce**:
As mentioned above. Should be very easy to reproduce: input any Chinese or Japanese words in Tabby and then use left-arrow and right-arrow keys to move cursor , see if the "ghost-like" vertical lines shows up.

**Comment by Eugeny** [comment]
Thanks for the report, does it also show up for the same characters in other fonts?

**Comment by MintYale** [comment]
> Thanks for the report, does it also show up for the same characters in other fonts?

YES. I've tried [Consolas, JetBrains Mono, Monaco] and some more, they all end up with the "ghost vertical lines" as long as you use left/right-arrow keys.  And sometimes after Ctrl＋L(clear screen), the vertical lines still stays there on the screen alone.

**Comment by MintYale** [comment]
> Thanks. Does enabling/disabling WebGL in Settings -> Terminal -> Frontend and reopening the tab have any effect? Do other machines show the same behaviour?

Unfortunately enabling/disabling WebGL in Settings -> Terminal -> Frontend and reopening the tab doesn't work on this issue. I've tried. You can reproduce and test this by coping/pasting any Chinese words such as "中文输入测试" in Tabby-v1.0.177(x64.exe Edition) and then move left/right-arrow keys, the "ghost vertical lines" shows up at 100% chance. Maybe you should check what's the different/unique font-display-strategy in Tabby from the others, 'cause other terminal apps doesn't has this issue.

**Comment by Eugeny** [comment]
Still unable to reproduce unfortunately, hence the question whether other machines show the same behaviour.

**Comment by SekiBetu** [comment]
> > Thanks for the report, does it also show up for the same characters in other fonts?
> 
> YES. I've tried [Consolas, JetBrains Mono, Monaco] and some more, they all end up with the "ghost vertical lines" as long as you use left/right-arrow keys. And sometimes after Ctrl＋L(clear screen), the vertical lines still stays there on the screen alone.

which version are you using, the latest version can't even find the JetBrains Mono font

**Comment by MintYale** [comment]
YES. I've tried [Consolas, JetBrains Mono,Monaco] and more, they all end up with the "ghost vertical lines". And sometimes after Ctrl＋L(clear screen), the vertical lines still stays there alone.

发自我的iPhone


------------------ Original ------------------
From: Eugeny ***@***.***&gt;
Date: Mon,May 23,2022 1:46 AM
To: Eugeny/tabby ***@***.***&gt;
Cc: MintYale ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [Eugeny/tabby] A kind of Non-English Characters Display Error (Issue #6479)





 
Thanks for the report, does it also show up for the same characters in other fonts?

**Comment by Eugeny** [comment]
Do you see any difference when changing the frontend to WebGL and back under Settings -> Terminal? (you need to open a new tab for the setting to take effect)

**Comment by MintYale** [comment]
I found that this strange problem has been solved somehow at some point. It's all good for CJK characters now. (tested on latest version 1.0.223, there is no ghost-lines anymore). The issue should be closed I think. Thanks for your work.
